sdk/go/common/resource/urn_test.go

```diff
@@ -85,3 +88,147 @@ func TestIsValid(t *testing.T) {
 		assert.True(t, urn.IsValid(), "IsValid expected to be true: %v", urn)
 	}
 }
+
+func TestParseURN(t *testing.T) {
+	t.Parallel()
+
+	t.Run("Positive Tests", func(t *testing.T) {
+		t.Parallel()
+
+		goodUrns := []string{
+			"urn:pulumi:test::test::pulumi:pulumi:Stack::test-test",
+			"urn:pulumi:stack-name::project-name::my:customtype$aws:s3/bucket:Bucket::bob",
+			"urn:pulumi:stack::project::type::",
+			"urn:pulumi:stack::project::type::some really ::^&\n*():: crazy name",
+			"urn:pulumi:stack::project with whitespace::type::some name",
+		}
+		for _, str := range goodUrns {
+			urn, err := ParseURN(str)
+			assert.NoErrorf(t, err, "Expecting %v to parse as a good urn", str)
+			assert.NotEmptyf(t, urn, "Expecting non empty urn parsing %v", str)
```

Comment:
Shouldn't this just equals check that it's the same as the input but now an URN?